<a href="https://colab.research.google.com/github/michalhal92/coffee-store-sales/blob/main/Coffee_Shop_Sales.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Wczytanie danych z pliku Excel, stworzenie DataFrame

In [3]:
import pandas as pd

baza = pd.read_excel("/content/sample_data/Coffe_sales.xlsx")

df = pd.DataFrame(baza)

print(df)

           date                datetime  hour_of_day cash_type  \
0    2024-03-01 2024-03-01 10:15:50.520           10      card   
1    2024-03-01 2024-03-01 12:19:22.539           12      card   
2    2024-03-01 2024-03-01 12:20:18.089           12      card   
3    2024-03-01 2024-03-01 13:46:33.006           13      card   
4    2024-03-01 2024-03-01 13:48:14.626           13      card   
...         ...                     ...          ...       ...   
3631 2025-03-23 2025-03-23 10:34:54.894           10      card   
3632 2025-03-23 2025-03-23 14:43:37.362           14      card   
3633 2025-03-23 2025-03-23 14:44:16.864           14      card   
3634 2025-03-23 2025-03-23 15:47:28.723           15      card   
3635 2025-03-23 2025-03-23 18:11:38.635           18      card   

                     card  money    coffee_name Time_of_Day Weekday  \
0     ANON-0000-0000-0001  38.70          Latte     Morning     Fri   
1     ANON-0000-0000-0002  38.70  Hot Chocolate   Afternoon     F

# Szybka analiza tabeli

In [4]:
# Ilość wierszy, nazwy kolumn, typy danych z info o not-nullach
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3636 entries, 0 to 3635
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   date         3636 non-null   datetime64[ns]
 1   datetime     3636 non-null   datetime64[ns]
 2   hour_of_day  3636 non-null   int64         
 3   cash_type    3636 non-null   object        
 4   card         3547 non-null   object        
 5   money        3636 non-null   float64       
 6   coffee_name  3636 non-null   object        
 7   Time_of_Day  3636 non-null   object        
 8   Weekday      3636 non-null   object        
 9   Month_name   3636 non-null   object        
 10  Weekdaysort  3636 non-null   int64         
 11  Monthsort    3636 non-null   int64         
dtypes: datetime64[ns](2), float64(1), int64(3), object(6)
memory usage: 341.0+ KB


In [5]:
# Info o nullach
df.isnull().sum()

date            0
datetime        0
hour_of_day     0
cash_type       0
card           89
money           0
coffee_name     0
Time_of_Day     0
Weekday         0
Month_name      0
Weekdaysort     0
Monthsort       0
dtype: int64

In [6]:
# Procent brakujących danych
print("\nProcent braków:\n", (df.isnull().mean() * 100).round(2))
# 2.45% nulli w cards wygląda na mało istotne


Procent braków:
 date           0.00
datetime       0.00
hour_of_day    0.00
cash_type      0.00
card           2.45
money          0.00
coffee_name    0.00
Time_of_Day    0.00
Weekday        0.00
Month_name     0.00
Weekdaysort    0.00
Monthsort      0.00
dtype: float64


In [7]:
# Podstawowe statystyki numeryczne dla kolumn numerycznych
df.describe()

date                       datetime  \
count                           3636                           3636   
mean   2024-09-30 11:56:02.376237568  2024-10-01 02:35:30.535053568   
min              2024-03-01 00:00:00     2024-03-01 10:15:50.520000   
25%              2024-07-03 00:00:00  2024-07-03 16:54:06.084750080   
50%              2024-10-06 12:00:00  2024-10-07 02:55:12.649500160   
75%              2025-01-08 00:00:00  2025-01-08 07:55:20.299750144   
max              2025-03-23 00:00:00     2025-03-23 18:11:38.635000   
std                              NaN                            NaN   

       hour_of_day        money  Weekdaysort    Monthsort  
count  3636.000000  3636.000000  3636.000000  3636.000000  
mean     14.166942    31.746859     3.847910     6.394664  
min       6.000000    18.120000     1.000000     1.000000  
25%      10.000000    27.920000     2.000000     3.000000  
50%      14.000000    32.820000     4.000000     6.000000  
75%      18.000000    35.760000     6.000000    10.000000  
max      22.000000    40.000000     7.000000    12.000000  
std       4.227755     4.919926     1.975983     3.480686

In [8]:
# Liczba unikalnych wpisów w poszczególnych kolumnach - pokazuje różnorodność wartości
df.nunique()

date            381
datetime       3636
hour_of_day      17
cash_type         2
card           1316
money            20
coffee_name       8
Time_of_Day       3
Weekday           7
Month_name       12
Weekdaysort       7
Monthsort        12
dtype: int64

*   date: format 2024-03-01, rozpiętośc dat od 2024-03-01 do 2025-03-23
*   datetime: format 2024-03-01 10:15:50.520 - data z godziną

In [9]:
# Sprawdzenie wartosci hour_of_day
print(f"W kolumnie 'hour_of_day' wystepują unikalne wartości:")
print(df['hour_of_day'].value_counts().sort_index())
# ok, wszystkie w ligocznym zaresie

W kolumnie 'hour_of_day' wystepują unikalne wartości:
hour_of_day
6       5
7      91
8     235
9     247
10    349
11    294
12    249
13    227
14    228
15    245
16    282
17    245
18    220
19    235
20    171
21    197
22    116
Name: count, dtype: int64


In [10]:
# Sprawdzenie wartosci card, która jest stringiem,
print("W kolumnie 'card' wystepują unikalne wartości:")
print(df['card'].value_counts(dropna=False))
# schemat ANON-xxxx-xxxx-xxxx, występują stali klienci

W kolumnie 'card' wystepują unikalne wartości:
card
ANON-0000-0000-0012    129
ANON-0000-0000-0141    108
NaN                     89
ANON-0000-0000-1163     77
ANON-0000-0000-0276     69
                      ... 
ANON-0000-0000-1300      1
ANON-0000-0000-1299      1
ANON-0000-0000-1296      1
ANON-0000-0000-0015      1
ANON-0000-0000-0014      1
Name: count, Length: 1317, dtype: int64


In [11]:
# Sprawdzenie wartosci money
print(f"W kolumnie 'money' wystepują unikalne wartości:")
print(df['money'].value_counts().sort_index())
# wartości liczbowe do 2 liczb po przecinku

W kolumnie 'money' wystepują unikalne wartości:
money
18.12      36
21.06      59
23.02     208
24.00      14
25.00       4
25.96     459
27.92     334
28.90      93
29.00       8
30.00      11
30.86     389
32.82     490
33.80      51
34.00       7
35.00       8
35.76    1014
37.72     273
38.70     128
39.00      18
40.00      32
Name: count, dtype: int64


In [12]:
# Sprawdzenie wartosci coffee_name, która jest stringiem,
print("W kolumnie 'coffee_name' wystepują unikalne wartości:")
print(df['coffee_name'].value_counts(dropna=False))
# o różnych napojów, nazewnictwo jednolite

W kolumnie 'coffee_name' wystepują unikalne wartości:
coffee_name
Americano with Milk    824
Latte                  782
Americano              578
Cappuccino             501
Cortado                292
Hot Chocolate          282
Cocoa                  243
Espresso               134
Name: count, dtype: int64


In [13]:
print("W kolumnie 'Time_of_Day' wystepują unikalne wartości:")
print(df['Time_of_Day'].value_counts(dropna=False))

W kolumnie 'Time_of_Day' wystepują unikalne wartości:
Time_of_Day
Afternoon    1231
Morning      1221
Night        1184
Name: count, dtype: int64


In [14]:
print("W kolumnie 'Weekday' wystepują unikalne wartości:")
print(df['Weekday'].value_counts(dropna=False))

W kolumnie 'Weekday' wystepują unikalne wartości:
Weekday
Tue    585
Mon    561
Fri    544
Thu    520
Wed    510
Sat    482
Sun    434
Name: count, dtype: int64


In [15]:
print(f"W kolumnie 'Weekdaysort' wystepują unikalne wartości:")
print(df['Weekdaysort'].value_counts().sort_index())

W kolumnie 'Weekdaysort' wystepują unikalne wartości:
Weekdaysort
1    561
2    585
3    510
4    520
5    544
6    482
7    434
Name: count, dtype: int64


In [16]:
print("W kolumnie 'Month_name' wystepują unikalne wartości:")
print(df['Month_name'].value_counts(dropna=False))

W kolumnie 'Month_name' wystepują unikalne wartości:
Month_name
Mar    525
Oct    426
Feb    423
Sep    344
Aug    272
May    267
Dec    259
Nov    259
Jul    237
Jun    227
Jan    201
Apr    196
Name: count, dtype: int64


In [17]:
print(f"W kolumnie 'Monthsort' wystepują unikalne wartości:")
print(df['Monthsort'].value_counts().sort_index())

W kolumnie 'Monthsort' wystepują unikalne wartości:
Monthsort
1     201
2     423
3     525
4     196
5     267
6     227
7     237
8     272
9     344
10    426
11    259
12    259
Name: count, dtype: int64
